owltorrent-network/src/main/java/edu/rice/owltorrent/network/TorrentRepositoryImpl.java

```diff
@@ -0,0 +1,39 @@
+package edu.rice.owltorrent.network;
+
+import edu.rice.owltorrent.common.entity.TwentyByteId;
+import java.util.Optional;
+import java.util.concurrent.ConcurrentHashMap;
+import lombok.extern.log4j.Log4j2;
+
+@Log4j2(topic = "general")
+public class TorrentRepositoryImpl implements TorrentRepository {
+  private ConcurrentHashMap<TwentyByteId, TorrentManager> infoHashToTorrentManager;
```

Comment:
make this final. nit: Also can inline the call to new and have an empty constructor.